File: src/lib/pjecalc/core/comum/validators/greater-than-validator.ts
/**
 * PJe-Calc v2.15.1 — GreaterThanValidator
 * Porte de: br.jus.trt8.pjecalc.negocio.comum.validators.GreaterThanValidator
 */
import Decimal from 'decimal.js';
import { HelperDate } from '../../base/comum/helper-date';
import { CustomValidator } from './custom-validator';
import type { GreaterThan_ } from './greater-than';
import type { ValidatorContext } from './validator-context';

export class GreaterThanValidator extends CustomValidator<GreaterThan_> {
  private static readonly COMPARED_KEY = '1';

  getMessage(): string { return this.spec.message; }
  getGroups(): number[] { return this.spec.groups; }
  protected getCondition(): string { return this.spec.condition; }

  getParameter(key: string): unknown {
    if (key === GreaterThanValidator.COMPARED_KEY) return this.spec.value;
    return null;
  }

  isValid(context: ValidatorContext): boolean {
    const value1 = context.getValue();
    const value2 = context.getMemberValue(this.spec.value);
    if (value1 == null || value2 == null) return true;
    if (value1 instanceof Date && value2 instanceof Date) {
      return HelperDate.dateBefore(value2, value1);
    }
    if (value1 instanceof Decimal && value2 instanceof Decimal) {
      return value1.greaterThan(value2);
    }
    return true;
  }
}

File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/comum/validators/GreaterThanValidator.java
/*
 * Decompiled with CFR 0.152.
 */
package br.jus.trt8.pjecalc.negocio.comum.validators;

import br.jus.trt8.pjecalc.base.comum.HelperDate;
import br.jus.trt8.pjecalc.base.comum.Utils;
import br.jus.trt8.pjecalc.base.comum.validadores.CustomValidator;
import br.jus.trt8.pjecalc.base.comum.validadores.ValidatorContext;
import br.jus.trt8.pjecalc.negocio.comum.validators.GreaterThan;
import java.math.BigDecimal;
import java.util.Date;

public class GreaterThanValidator
extends CustomValidator<GreaterThan> {
    private static final String COMPARED_KEY = "1";

    @Override
    public Object getParameter(String key) {
        if (key.equals(COMPARED_KEY)) {
            return ((GreaterThan)this.getAnnotation()).value();
        }
        return null;
    }

    @Override
    public String getMessage() {
        return ((GreaterThan)this.getAnnotation()).message();
    }

    @Override
    public byte[] getGroups() {
        return ((GreaterThan)this.getAnnotation()).groups();
    }

    @Override
    protected String getCondition() {
        return ((GreaterThan)this.getAnnotation()).condition();
    }

    @Override
    public boolean isValid(ValidatorContext context) {
        Object value1 = context.getValue();
        Object value2 = context.getMemberValue(((GreaterThan)this.getAnnotation()).value());
        if (Utils.nulo(value1) || Utils.nulo(value2)) {
            return true;
        }
        if (value1 instanceof Date && value2 instanceof Date) {
            return HelperDate.dateBefore((Date)value2, (Date)value1);
        }
        if (value1 instanceof BigDecimal && value2 instanceof BigDecimal) {
            return ((BigDecimal)value1).compareTo((BigDecimal)value2) > 0;
        }
        return true;
    }
}

